src/daft-csv/src/read.rs

```diff
@@ -297,6 +379,103 @@ async fn read_csv_single_into_table(
     }
 }
 
+async fn read_csv_single_into_table_stream(
```

Comment:
This function is basically `read_csv_single_into_table` without the `try_collect`